docs/DeveloperGuide.md

```diff
@@ -1034,19 +1034,36 @@ otherwise)
 
 ### Non-Functional Requirements
 
+**Technical requirements:**
 1. The application should be platform-independent and should run on all major operating systems as long as it has
    Java `11` or above installed.
-2. Should be able to hold up to 1000 persons without a noticeable sluggishness in performance for typical usage.
-3. A user with above average typing speed for regular English text (i.e. not code, not system admin commands) should be
+2. The system should work on both 32-bit and 64-bit environments.
+
+**Performance requirements:**
+1. Should be able to hold up to 1000 Students/Groups without a noticeable sluggishness in performance for typical usage.
+2. ProfBook should be able to respond in less than 0.5 seconds.
+
+**Quality requirements:**
+1. A user with above average typing speed for regular English text (i.e. not code, not system admin commands) should be
    able to accomplish most of the tasks faster using commands than using the mouse.
+2. The system can be used for basic function without reading the user guide.
 
-*{More to be added}*
+**Constraints:**
+1. Should work without an internet connection.
+2. Should be able to support frequent updating of data.
 
 ### Glossary
```

Comment:
I think glossary can add more, can consider looking at this
![image](https://github.com/AY2324S1-CS2103T-W15-2/tp/assets/94535035/f67f187b-207e-43de-a437-d992a59fbdde)

Reply:
updated with more